### PR TITLE
Add support for reading EMPAD files.

### PR DESCRIPTION
**Pull request by ericpre**
It needs smallest possible test file to be ready for review. If someone can provide some, this would be helpful. In the mean time, I open this PR, in case it can be of interested.

### Progress of the PR
- [x]  Add EMPAD reader,
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

**Comment by woozey** [comment]
Smallest I have is 66.6 MB, let me know if you want to have this one.

**Comment by ericpre** [comment]
@woozey, instead of adding a test file, it is now creating the raw data before the test and only the xml file is original.
The data I have are not calibrated, so I am not sure if reading the calibration of the signal dimension is working correctly.
Could you please share some of your file? So I can checked if there are any difference in the xml file, etc. Are they calibrated (signal dimension)?

File: hyperspy/io_plugins/empad.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            'record-by':'image'}   
+    if om.has_item('root.count'):
+        # Stack of images
+        info.update({'series_count':int(om.root.scan_parameters.series_count)})
+    elif om.has_item('root.pix_x') and om.has_item('root.pix_y'):
+        # 2D x 2D
+        info.update({'scan_x':int(om.root.pix_x),
+                     'scan_y':int(om.root.pix_y)})
+    else:
+        raise IOError("Unsupported Empad file: the scan parameters can not "
```
Comment (tjof2) [inline hyperspy/io_plugins/empad.py]: "cannot" rather than "can not" :)

File: hyperspy/io_plugins/empad.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
```
Comment (dnjohnstone) [inline hyperspy/io_plugins/empad.py]: 2007-2020

**Review by dnjohnstone — APPROVED** [review]
This basically looks good to me - a stupid thing about dates, but otherwise ok and would be good to have in. 

Sorting the merge conflicts and then verifying that the tests are ok (current failures are due to the old tifffile issue) would make this ready to merge for me.

**Review by tjof2 — APPROVED** [review]

**Comment by ericpre** [comment]
@tjof2, if you are happy with this PR, can you please merge it! ;)

**Comment by francisco-dlp** [comment]
Why this doesn't have an entry in the User Guide?

**Comment by ericpre** [comment]
It does! It was missing from this PR but it has been rectified in https://github.com/hyperspy/hyperspy/pull/2439.

**Comment by francisco-dlp** [comment]
Great, thanks!

**Pull request merged**